libcloud/compute/drivers/cloudstack.py

```diff
@@ -4652,7 +4652,8 @@ def _to_node(self, data, public_ips=None):
             extra['tags'] = self._get_resource_tags(data['tags'])
 
         node = CloudStackNode(id=id, name=name, state=state,
-                              public_ips=public_ips, private_ips=private_ips,
+                              public_ips=list(set(public_ips)),
```

Comment:
The issue only exists for public ips because private ips are not passed with argument.
The issue occurs when an ip is included in both `data['nic']` and argument `public_ips`.